internal/services/mobilenetwork/mobile_network_data_network_resource.go

```diff
@@ -0,0 +1,241 @@
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: mobilenetwork.ValidateMobileNetworkID,
+		},
+
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": commonschema.Location(),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r DataNetworkResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r DataNetworkResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model DataNetworkModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.MobileNetwork.DataNetworkClient
+			mobileNetworkId, err := mobilenetwork.ParseMobileNetworkID(model.MobileNetworkMobileNetworkId)
+			if err != nil {
+				return err
+			}
+
+			id := datanetwork.NewDataNetworkID(mobileNetworkId.SubscriptionId, mobileNetworkId.ResourceGroupName, mobileNetworkId.MobileNetworkName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := datanetwork.DataNetwork{
+				Location:   location.Normalize(model.Location),
+				Properties: &datanetwork.DataNetworkPropertiesFormat{},
+				Tags:       &model.Tags,
+			}
+
+			if model.Description != "" {
+				properties.Properties.Description = &model.Description
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r DataNetworkResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.DataNetworkClient
+
+			id, err := datanetwork.ParseDataNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model DataNetworkModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", id)
+			}
+
+			properties := *resp.Model
+
+			if metadata.ResourceData.HasChange("description") {
+				if model.Description != "" {
+					properties.Properties.Description = &model.Description
+				} else {
+					properties.Properties.Description = nil
+				}
+			}
+
+			properties.SystemData = nil
```

Comment:
we don't need to do this, `hashicorp/go-azure-helpers` takes care of this for us: https://github.com/hashicorp/go-azure-helpers/blob/0bf5b6421fcc4dd925781cab92ada71e2ba03009/resourcemanager/systemdata/model.go#L19-L23

```suggestion
```